src/tracer.c

```diff
@@ -124,6 +124,7 @@ static bool next_cf_insn(vmi_instance_t vmi, addr_t start)
     if ( VMI_FAILURE == vmi_read(vmi, &ctx, TRACER_BUFFER_SIZE, buff, &read) )
     {
         if ( debug ) printf("Failed to grab memory from 0x%lx with PT 0x%lx\n", start, target_pagetable);
+        crash = 1; // May be the result of a currupt branch
```

Comment:
So if the RIP is now pointing to unmapped memory there could be a couple reasons for that. For example, we could be landing in a memory that's paged out. But we could also be transitioning to a different pagetable and I see that above we are always trying to fetch the memory from the `target_pagetable` that we started with. Perhaps this function just needs to be updated to take the `dtb` as well instead of always using the same. Even after all those adjustments if this catch fetch memory I wouldn't report a crash since if its truly going to point to memory that's not present it will crash in a way that will be caught by the already trapped sink points.